<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>LayerSelectorDialog</class>
 <widget class="QDialog" name="LayerSelectorDialog">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>700</width>
    <height>600</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Layer Selection</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <item>
    <widget class="QGroupBox" name="filterGroup">
     <property name="title">
      <string>Layer Filtering</string>
     </property>
     <layout class="QHBoxLayout" name="filterLayout">
      <item>
       <widget class="QLabel" name="geometryTypeLabel">
        <property name="text">
         <string>Geometry Type:</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QComboBox" name="geometryTypeCombo">
        <item>
         <property name="text">
          <string>All Layers</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Point Layers (Junctions/Manholes)</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Line Layers (Pipes/Conduits)</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Polygon Layers</string>
         </property>
        </item>
       </widget>
      </item>
      <item>
       <spacer name="filterSpacer">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>40</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
      <item>
       <widget name="refreshButton" class="QPushButton">
        <property name="text">
         <string>Refresh</string>
        </property>
        <property name="toolTip">
         <string>Refresh the layer list</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QSplitter" name="splitter">
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
     <widget class="QGroupBox" name="layerListGroup">
      <property name="title">
       <string>Available Layers</string>
      </property>
      <layout class="QVBoxLayout" name="layerListLayout">
       <item>
        <widget class="QListWidget" name="layerListWidget">
         <property name="toolTip">
          <string>Select a layer to view its information and sample data</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QGroupBox" name="layerInfoGroup">
      <property name="title">
       <string>Layer Information</string>
      </property>
      <layout class="QVBoxLayout" name="layerInfoLayout">
       <item>
        <widget class="QLabel" name="layerNameLabel">
         <property name="text">
          <string>Select a layer to view information</string>
         </property>
         <property name="wordWrap">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QGroupBox" name="layerDetailsGroup">
         <property name="title">
          <string>Layer Details</string>
         </property>
         <layout class="QFormLayout" name="layerDetailsLayout">
          <item row="0" column="0">
           <widget class="QLabel" name="geometryTypeDetailLabel">
            <property name="text">
             <string>Geometry Type:</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QLabel" name="geometryTypeValueLabel">
            <property name="text">
             <string>-</string>
            </property>
           </widget>
          </item>
          <item row="1" column="0">
           <widget class="QLabel" name="featureCountLabel">
            <property name="text">
             <string>Feature Count:</string>
            </property>
           </widget>
          </item>
          <item row="1" column="1">
           <widget class="QLabel" name="featureCountValueLabel">
            <property name="text">
             <string>-</string>
            </property>
           </widget>
          </item>
          <item row="2" column="0">
           <widget class="QLabel" name="crsLabel">
            <property name="text">
             <string>CRS:</string>
            </property>
           </widget>
          </item>
          <item row="2" column="1">
           <widget class="QLabel" name="crsValueLabel">
            <property name="text">
             <string>-</string>
            </property>
            <property name="wordWrap">
             <bool>true</bool>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QGroupBox" name="fieldsGroup">
         <property name="title">
          <string>Layer Fields</string>
         </property>
         <layout class="QVBoxLayout" name="fieldsLayout">
          <item>
           <widget class="QTableWidget" name="fieldsTableWidget">
            <property name="toolTip">
             <string>Available fields in the selected layer</string>
            </property>
            <property name="alternatingRowColors">
             <bool>true</bool>
            </property>
            <property name="selectionBehavior">
             <enum>QAbstractItemView::SelectRows</enum>
            </property>
            <property name="sortingEnabled">
             <bool>true</bool>
            </property>
            <column>
             <property name="text">
              <string>Field Name</string>
             </property>
            </column>
            <column>
             <property name="text">
              <string>Type</string>
             </property>
            </column>
            <column>
             <property name="text">
              <string>Length</string>
             </property>
            </column>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QGroupBox" name="sampleDataGroup">
         <property name="title">
          <string>Sample Data (First 5 Features)</string>
         </property>
         <layout class="QVBoxLayout" name="sampleDataLayout">
          <item>
           <widget class="QTableWidget" name="sampleDataTableWidget">
            <property name="toolTip">
             <string>Sample data from the selected layer</string>
            </property>
            <property name="alternatingRowColors">
             <bool>true</bool>
            </property>
            <property name="selectionBehavior">
             <enum>QAbstractItemView::SelectRows</enum>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
      </layout>
     </widget>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="validationGroup">
     <property name="title">
      <string>Layer Validation</string>
     </property>
     <layout class="QVBoxLayout" name="validationLayout">
      <item>
       <widget class="QLabel" name="validationStatusLabel">
        <property name="text">
         <string>Select a layer to see validation status</string>
        </property>
        <property name="wordWrap">
         <bool>true</bool>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QDialogButtonBox" name="buttonBox">
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
     <property name="standardButtons">
      <set>QDialogButtonBox::Cancel|QDialogButtonBox::Ok</set>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>buttonBox</sender>
   <signal>accepted()</signal>
   <receiver>LayerSelectorDialog</receiver>
   <slot>accept()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>248</x>
     <y>254</y>
    </hint>
    <hint type="destinationlabel">
     <x>157</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonBox</sender>
   <signal>rejected()</signal>
   <receiver>LayerSelectorDialog</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>316</x>
     <y>260</y>
    </hint>
    <hint type="destinationlabel">
     <x>286</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>